### PR TITLE
Reduced verbosity of error logging for remote WPS inputs.

### PR DESCRIPTION
**Pull request by sikeoka**
This pull request updates error handling for loading remote WPS inputs to be similar to how org.geoserver.wms.map.GetMapKvpRequestReader handles errors when loading an SLD from a URL.

This pull request can be backported to 2.12.x and 2.13.x.

### CODE REVIEW

File: src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/RemoteRequestInputProvider.java [hunk 1]
```diff
@@ -204,10 +204,29 @@ protected Object getValueInternal(ProgressListener listener) throws Exception {
...
+                if (result == null || complexPPIO.getType().isInstance(result)) {
+                    return result;
+                }
+                // Some text parsers return a Map when it can not be converted to
+                // the proper type.  Detect those errors here rather than later.
+                throw new IllegalArgumentException(
+                        "Decoded result is not a "
+                                + complexPPIO.getType().getName()
+                                + ", got a: "
+                                + result);
```
Comment (bradh) [inline src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/RemoteRequestInputProvider.java]: Is this trying to display `result` or should it be showing the type of `result`?

File: src/extension/wps/wps-core/src/test/java/org/geoserver/wps/my-secret.xml [hunk 2]
```diff
@@ -0,0 +1 @@
+<foo>MY_SECRET</foo>
```
Comment (bradh) [inline src/extension/wps/wps-core/src/test/java/org/geoserver/wps/my-secret.xml]: If you are going to make other changes to this PR, can you add the newline at the end of this file too?

File: src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/RemoteRequestInputProvider.java (hunk 1 above)
Comment (sikeoka) [reply src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/RemoteRequestInputProvider.java]: I changed it to show the type of result.

File: src/extension/wps/wps-core/src/test/java/org/geoserver/wps/my-secret.xml (hunk 2 above)
Comment (sikeoka) [reply src/extension/wps/wps-core/src/test/java/org/geoserver/wps/my-secret.xml]: I added the newline at the end of this file.

**Pull request merged**